paddle/fluid/operators/math/softmax_impl.h

```diff
@@ -38,16 +40,37 @@ template <typename DeviceContext, typename T, bool is_test, typename Enable>
 void SoftmaxFunctor<DeviceContext, T, is_test, Enable>::operator()(
     const DeviceContext& context, const int axis_dim,
     const framework::Tensor* X, framework::Tensor* Y) {
-  auto logits = EigenMatrix<T>::From(*X);
-  auto softmax = EigenMatrix<T>::From(*Y);
-
+  auto in_dims = X->dims();
   const int kBatchDim = 0;
   const int kClassDim = 1;
 
-  const int batch_size = logits.dimension(kBatchDim);
-  const int num_classes = logits.dimension(kClassDim);
+  const int num_classes = in_dims[kClassDim];
+  const int batch_size = in_dims[kBatchDim];
   const int num_remain = num_classes / axis_dim;
```

Comment:
axis_dim can be (to my understanding): [0, rank-1] . So it may happen that  there is exception here when dividing by zero . If my understanding is wrong then please explain?

Reply:
https://github.com/PaddlePaddle/Paddle/blob/dfdcd91869595c7f068767e15f05f325d8a1de9c/paddle/fluid/operators/softmax_op.h#L56

It's from here, so could no be zero.

Reply:
Ok, makes sense